seed_isort_config.py

```diff
@@ -101,7 +110,16 @@ def main(argv=None):
             'one of {}...'.format(prefix, ', '.join(SUPPORTED_CONF_FILES)),
         )
 
-        with io.open('.isort.cfg', mode, encoding='UTF-8') as f:
+        # Make sure the target directory exists before attempting to
+        # write to it. Catch the OSError to mimic the behavior of the
+        # ``exist_ok`` argument, which is supported in Python 3.2+ but
+        # not 2.7.
+        try:
+            os.makedirs(args.settings_path)
+        except OSError:
+            pass
```

Comment:
let's be explicit here:

```python
    if not os.path.isdir(args.settings_path):
        raise
```

also let's adjust the test so it hits the case where the directory doesn't exist (right now I think we're hitting the "always exists" case)

Reply:
Do you have an opinion on how to fail gracefully here? e.g.

```python
if not os.path.isdir(args.settings_path):
    sys.stderr.write(
        '{} is not a directory or does not exist. '
        'Could not create {}.'.format(args.settings_path, filename)
    )
    sys.exit(1)
```

Reply:
let's just `raise`, the target audience is programmers anyway and it's such an edgecase it's not worth testing or supporting.  In both python2 and python3 the stacktrace is helpful enough (mentions the path that was attempted to be created and "File exists") so I think that's good enough

Reply:
Is this coming around to the original error? Without any exception handling or checking, this is the output

Reply:
- `makedirs('path/to/file')` is incredibly unlikely
- `open('path/to/file', 'w')` where `path/to` doesn't exist is much more likely

Also the `makedirs(...)` => "is a file" is a much more obvious error than `open(..., 'w')` "No such file or directory" (of course it doesn't exist! that's why I'm trying to write it! oh you mean the directory 😭)

Reply:
And the former isn't fixable without a human, whereas the latter is

Reply:
just empty `raise` ([I wrote the code for you!](https://github.com/asottile/seed-isort-config/pull/11/files#r203572494)), it'll reraise the existing error